docs/TektonConfig.md

```diff
@@ -408,3 +426,84 @@ platforms:
+There is a filed called `options` available in all the components.<br>
+
+>**NOTE:** There is a possibility to have two different values for a field.<br> 
+An example: with a pre-defined field you can set value and the same filed may be defined under `options` as well. In that case value from `options` will be final.
+
+`options` filed is defined as follows,
+```yaml
+options:
+  disabled: false
+  configMaps:
+    config-leader-election:
+      metadata:
+        labels:
+        annotations:
+      data:
+        buckets: "2"
+    my-custom-config-map: # creates new configMap under targetNamespace
+      data:
+        foo: bar
+  deployments:
+    tekton-pipelines-controller:
+      metadata:
+        labels:
+          custom-label: "foo"
+        annotations:
+          custom-annotation: "foo"
+      spec:
+        replicas: 2
+        template:
+          spec:
+            affinity:
+            nodeSelector:
+            tolerations:
+            topologySpreadConstraints:
+            volumes:
+              - name: config-logging
+                configMap:
+                  name: pipeline-config-logging
+            initContainers:
+            containers:
+              - name: tekton-pipelines-controller
+                volumeMounts:
+                env:
+                  - name: CONFIG_LOGGING_NAME
+                    value: pipeline-config-logging
+                args:
+                  - "--disable-ha=false"            
+```
+* `disabled` - disables the additional `options` support, if `disabled` set as `true`. default: `false`
+
+#### ConfigMaps
+Supports to update existing configMap also supports to create new configMap.
+
+The following fields are supported in `configMap`
+* `metadata`
+  * `labels` - supports add and update
+  * `annotations` - supports add and update
+* `data` - supports add and update
+
+#### Deployments
+Supports to update the existing deployments. But not supported to create new deployment.
+
+The following fields are supported in `deployment`
+* `metadata`
+  * `labels` - supports add and update
+  * `annotations` - supports add and update
+* `spec`
+  * `replicas` - updated deployment replicas count
+  * `template`
+    * `spec`
+      * `affinity` - replaces the existing Affinity with this, if not empty
+      * `nodeSelector` - replaces the existing NodeSelector with this, if not empty
+      * `tolerations` - replaces the existing tolerations with this, if not empty
+      * `topologySpreadConstraints` - replaces the existing TopologySpreadConstraints with this, if not empty
+      * `volumes` - adds and updates volumes
+      * `initContainers` - updates init-containers
+        * `resources` - replaces the resources requirements with this, if not empty
+        * `envs` - adds and updates environments
+        * `volumeMounts` - adds and updates VolumeMounts
+        * `args` - appends given args with existing arguments. **NOTE: THIS OPERATION DO NOT REPLACE EXISTING ARGS** 
```

Comment:
Should we change tha name to reflect that ?

Reply:
I have not setting the name explicitly. I have taken `ConfigMap` and `Deployment` types. If we need to change the name, we have to define our own schema/type for `ConfigMap` and `Deployment`

Reply:
Right, I see.